src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -87,6 +88,32 @@ class BasePaymentsPage extends React.Component {
             this.debounceSetShouldShowLoadingSpinner();
         }
 
+        if (this.state.shouldShowDefaultDeleteMenu || this.state.shouldShowPasswordPrompt) {
+            // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+            let shouldResetPaymentMethodData = false;
+
+            if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(this.props.bankAccountList[this.state.methodID])) {
+                // If the selected payment method is bank account and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(this.props.cardList[this.state.methodID])) {
+                // If the selected payment method is card and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && this.props.payPalMeData !== prevProps.payPalMeData && _.isEmpty(this.props.payPalMeData)) {
+                // If the selected payment method is paypal and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            }
+            if (shouldResetPaymentMethodData) {
+                // Close corresponding selected payment method modals which are open
+                if (this.state.shouldShowDefaultDeleteMenu) {
+                    this.hideDefaultDeleteMenu();
+                }
+                if (this.state.shouldShowPasswordPrompt) {
+                    this.hidePasswordPrompt();
+                }
+                this.resetSelectedPaymentMethodData();
```

Comment:
I noticed this redundant line previously as well. But didn't remove it because, it seems to be an extra protection to remove the selected method data is some edge the modal is closed without removing data. And it doesn't do any harm with redundant state change.

If we still want to remove this line, i can remove it.

Reply:
done